### PR TITLE
Xsiam firehose

### PR DESCRIPTION
**Pull request by jamesgreen-moj**
* added new xsiam firehose module
* Added ability to pull secrets in via data block
* Defined access_key & http_endpoint in secrets manager

- This is a draft waiting on endpoint details

### CODE REVIEW

File: modules/kinesis_firehose_xsiam/main.tf [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+resource "aws_iam_policy" "s3_kinesis_xsiam_policy" {
+
+  # Terraform's "jsonencode" function converts a
+  # Terraform expression result to valid JSON syntax.
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "s3:*"
```
Comment (smjmoj) [inline modules/kinesis_firehose_xsiam/main.tf:119]: This looks too permissive, could it be restricted?

File: modules/kinesis_firehose_xsiam/log_group_subscription.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "put_record" {
+  name_prefix = var.prefix
+  tags        = var.tags
+  policy      = <<-EOF
```
Comment (smjmoj) [inline modules/kinesis_firehose_xsiam/log_group_subscription.tf:31]: Is that a un wanted char on line 31?

**Review by smjmoj — COMMENTED** [review]
See queries.
Otherwise LGTM

File: modules/kinesis_firehose_xsiam/log_group_subscription.tf (hunk 2 above)
Comment (jamesgreen-moj) [reply modules/kinesis_firehose_xsiam/log_group_subscription.tf:31]: I stolen this code from a module that is already configured to do firehose actions:
https://github.com/observeinc/terraform-aws-kinesis-firehose/blob/93da8c06e516e4e30f186db0ae898b337b1f785e/main.tf#L158

What about it is incorrect?

File: modules/kinesis_firehose_xsiam/main.tf (hunk 1 above)
Comment (jamesgreen-moj) [reply modules/kinesis_firehose_xsiam/main.tf]: Updated to have more restricted permissions.

**Review by smjmoj — APPROVED** [review]
LGTM

File: modules/kinesis_firehose_xsiam/log_group_subscription.tf (hunk 2 above)
Comment (smjmoj) [reply modules/kinesis_firehose_xsiam/log_group_subscription.tf:31]: Looks wrong to me, but I've now looked that up and can see it's an intentional choice.
Learned something  new today

**Pull request merged**